subsystems/DiscordRPC/src/main/java/org/terasology/engine/subsystem/discordrpc/DiscordRPCSubSystem.java

```diff
@@ -38,281 +31,136 @@
-    private static final long DISCORD_APP_CLIENT_ID = 515274721080639504L;
-    private static final String DISCORD_APP_LARGE_IMAGE = "ss_6";
-    private static final int RECONNECT_TRIES = 5;
     private static DiscordRPCSubSystem instance;
 
-    private IPCClient ipcClient;
-    private boolean ready;
-    private boolean autoReconnect;
-    private Thread reconnectThread;
-    private RichPresence lastRichPresence;
-    private boolean reconnecting;
-    private int reconnectTries = 1;
-    private boolean connectedBefore;
     private Config config;
-    private String lastState;
-    private boolean dontTryAgain;
-    private boolean enabled;
+    private DiscordRPCThread thread;
 
     public DiscordRPCSubSystem() throws IllegalStateException {
         if (instance != null) {
             throw new IllegalStateException("More then one instance in the DiscordRPC");
         }
-        lastRichPresence = null;
-        ipcClient = new IPCClient(DISCORD_APP_CLIENT_ID);
-        ipcClient.setListener(this);
-        autoReconnect = true;
-        reconnectThread = new Thread(this);
-        reconnectThread.setName("DISCORD-RPC-RECONNECT");
-        reconnectThread.start();
-        instance = this;
-        enabled = false;
-        dontTryAgain = true;
-    }
 
-    public void sendRichPresence(RichPresence richPresence) {
-        this.lastRichPresence = richPresence;
-        if (!ready || lastRichPresence == null || !enabled) {
-            return;
-        }
-        ipcClient.sendRichPresence(lastRichPresence);
+        instance = this;
     }
 
     @Override
-    public void onReady(IPCClient client) {
-        if (reconnecting) {
-            logger.info("Discord RPC >> Reconnected!");
-            reconnectTries = 1;
-        } else {
-            logger.info("Discord RPC >> Connected!");
-            connectedBefore = true;
-        }
-        this.ipcClient = client;
-        if (!ready) {
-            ready = true;
-        }
-        if (lastRichPresence == null) {
-            RichPresence.Builder builder = new RichPresence.Builder();
-            builder.setLargeImage(DISCORD_APP_LARGE_IMAGE);
-            lastRichPresence = builder.build();
-        }
-        client.sendRichPresence(lastRichPresence);
-    }
+    public void initialise(GameEngine engine, Context rootContext) {
+        logger.info("Initializing...");
 
-    @Override
-    public void onDisconnect(IPCClient client, Throwable t) {
-        if (ready) {
-            ready = false;
+        thread = new DiscordRPCThread();
+        thread.getBuffer().setState("In Main Menu");
+
+        config = rootContext.get(Config.class);
+
+        if (config.getPlayer().isDiscordPresence()) {
+            thread.enable();
+        } else {
+            logger.info("Discord RPC is disabled! No connection is being made during initialization.");
+            thread.disable();
         }
-        logger.info("Discord RPC >> Disconnected!");
+        thread.start();
```

Comment:
Since we log the start of initializing as INFO, should we also load the success when the thread was started (analog to the case where the integration is disabled).

Reply:
The purpose of the thread is to be alive all the time until the game shuts down. But if the user disabled the RPC the thread will be in the sleeping mode until something changes and notifies it. When the thread gets notified it tries to connect to the IPC server. If it fails to connect 5 times at maximum then the thread will go sleep again.